aten/src/ATen/native/cuda/Loops.cuh

```diff
@@ -9,3 +47,195 @@
 #else
 #include <ATen/native/cuda/ROCmLoops.cuh>
 #endif
+
+namespace at { namespace native {
+
+// `needs_dynamic_casting` compares the types expected by iterator
```

Comment:
@ngimel Docs added here

Reply:
Cool, so it'll be good to go once you move out tests from Loops.cuh.

Reply:
Test moved to cuda_vectorized_test.cu